src/applications/vaos/new-appointment/components/ClinicChoicePage/useClinicFormState.jsx

```diff
@@ -0,0 +1,107 @@
+import useFormState from '../../../hooks/useFormState';
+import { getSiteIdFromFacilityId } from '../../../services/location';
+import { getClinicId } from '../../../services/healthcare-service';
+
+import {
+  getClinicsForChosenFacility,
+  getFormData,
+  selectPastAppointments,
+} from '../../redux/selectors';
+
+const initialSchema = {
+  type: 'object',
+  required: ['clinicId'],
+  properties: {
+    clinicId: {
+      type: 'string',
+      enum: [],
+    },
+  },
+};
+const uiSchema = {
+  clinicId: {
+    'ui:widget': 'radio',
+  },
+};
+
+export default function useClinicFormState() {
+  const initialData = useSelector(getFormData);
+  const clinics = useSelector(getClinicsForChosenFacility);
+  const pastAppointments = useSelector(selectPastAppointments);
+
+  const formState = useFormState({
+    initialSchema() {
+      let newSchema = initialSchema;
+      let filteredClinics = clinics;
+
+      if (pastAppointments) {
+        const pastAppointmentDateMap = new Map();
+        const siteId = getSiteIdFromFacilityId(initialData.vaFacility);
+
+        pastAppointments.forEach(appt => {
+          const apptTime = appt.startDate;
+          const latestApptTime = pastAppointmentDateMap.get(appt.clinicId);
+          if (
+            // Remove parse function when converting the past appointment call to FHIR service
+            appt.facilityId === siteId &&
+            (!latestApptTime || latestApptTime > apptTime)
+          ) {
+            pastAppointmentDateMap.set(appt.clinicId, apptTime);
+          }
+        });
+
+        filteredClinics = clinics.filter(clinic =>
+          // Get clinic portion of id
+          pastAppointmentDateMap.has(getClinicId(clinic)),
+        );
+      }
+
+      if (filteredClinics.length === 1) {
+        const clinic = filteredClinics[0];
+        newSchema = {
+          ...newSchema,
+          properties: {
+            clinicId: {
+              type: 'string',
+              title: `Would you like to make an appointment at ${
+                clinic.serviceName
+              }?`,
+              enum: [clinic.id, 'NONE'],
+              enumNames: [
+                'Yes, make my appointment here',
+                'No, I need a different clinic',
+              ],
+            },
+          },
+        };
+      } else {
+        newSchema = {
+          ...newSchema,
+          properties: {
+            clinicId: {
+              type: 'string',
+              title:
+                'Choose a clinic below or request a different clinic for this appointment.',
+              enum: filteredClinics.map(clinic => clinic.id).concat('NONE'),
+              enumNames: filteredClinics
+                .map(clinic => clinic.serviceName)
+                .concat('I need a different clinic'),
+            },
+          },
+        };
+      }
+
+      return newSchema;
+    },
+    uiSchema,
+    initialData,
+  });
+
+  return {
+    ...formState,
+    firstMatchingClinic: clinics?.find(
```

Comment:
Included this here so we don't need to pull the whole clinics list in the clinic page component as well as in this hook.